### PR TITLE
Add type declaration file tip in the documentation

### LINKED ISSUE #303: Where to add the vue-shim.d.ts file?
https://typescript.nuxtjs.org/guide/setup.html#configuration

> You will also need to provide types for Vue files by adding the following type declaration:

Where?

### PR DESCRIPTION
**Pull request by revolter**
Preview: https://deploy-preview-305--nuxt-typescript.netlify.com/guide/setup.html#configuration

Closes #303.

### CODE REVIEW

File: docs/guide/setup.md [hunk 1]
```diff
@@ -41,6 +41,10 @@ declare module "*.vue" {
 }
 ```
 
+::: tip
+You can place this file in the root directory of your project or a directory named `@types`. You can place it in a custom directory, but you'll need to configure [`typeRoots`](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html#types-typeroots-and-types) in the `tsconfig.json` file.
```
Comment (rchl) [inline docs/guide/setup.md]: > directory named `@types`

I know that this is what Kevin said on Discord, but I haven't seen any packages using `@types` directory. It's typically just `types`. `node_modules/@types/package-name` is a thing but that's not the same as having `@types` directory in your project.

I also usually put all this stuff in `./types` and typescript seems to pick those up without having to add them to `tsconfig.json` although documentation at https://www.typescriptlang.org/docs/handbook/tsconfig-json.html#types-typeroots-and-types doesn't seem to mention that being supported.

Comment (revolter) [reply docs/guide/setup.md]: So I should change `@types` to `types`? And if you want to name it `my-custom-types`, you still need that configuration, right?

Comment (kevinmarrec) [reply docs/guide/setup.md]: @rchl Yeah I think `@types` is kind of hidden auto detected folder by TypeScript, likely as `./types` I guess ?

Comment (rchl) [reply docs/guide/setup.md]: But then I would think it's common to use `types` directory while `@types` would look weird in a normal project.

Comment (revolter) [reply docs/guide/setup.md]: I changed it.

**Comment by kevinmarrec** [comment]
@rchl Does the current changes look good for you ?

**Comment by rchl** [comment]
Yes although I haven't verified that `You can place this file in the root directory of your project` is true.

**Comment by kevinmarrec** [comment]
@rchl I can confirm it works :)

**Comment by kevinmarrec** [comment]
Thanks for the PR @revolter

**Pull request merged**